e2e/analysis/analysis_test.go

```diff
@@ -15,123 +15,139 @@
 package analysis_test
 
 import (
+	"encoding/json"
 	"fmt"
+	"github.com/ZupIT/horusec-devkit/pkg/enums/languages"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
 
+	"github.com/google/uuid"
 	. "github.com/onsi/ginkgo"
+	ginkgoconfig "github.com/onsi/ginkgo/config"
 	. "github.com/onsi/gomega"
-	"github.com/onsi/gomega/gexec"
 
-	"github.com/ZupIT/horusec-devkit/pkg/enums/languages"
-	"github.com/ZupIT/horusec-devkit/pkg/enums/tools"
+	"github.com/ZupIT/horusec-devkit/pkg/utils/logger"
+	"github.com/ZupIT/horusec/config"
+	"github.com/ZupIT/horusec/e2e/analysis"
+	customimages "github.com/ZupIT/horusec/internal/entities/custom_images"
 	"github.com/ZupIT/horusec/internal/utils/testutil"
 )
 
-type AnalysisTestCase struct {
-	Language       languages.Language
-	AnalysisFolder string
-	ExpectedTools  []tools.Tool
-}
+const isWindows = runtime.GOOS == "windows"
+const isDarwin = runtime.GOOS == "darwin"
+const horusecConfigName = "horusec-config.json"
+
+var _ = Describe("Run a complete horusec analysis when build tools locally", func() {
+	tmpDir := CreateHorusecConfigAndReturnTMPDirectory()
+
+	Describe("Running e2e tests when build tools locally", func() {
+		allTestsCases := analysis.NewTestCase()
+
+		for idx := range allTestsCases {
+			if !isDarwin {
+				horusecConfigFilePathBuildLocally := filepath.Join(tmpDir, horusecConfigName)
+				RunTestCase(allTestsCases, horusecConfigFilePathBuildLocally, "(build locally)", idx)
+
+				time.Sleep(2 * time.Second)
+			}
 
-func (A AnalysisTestCase) RunAnalysisTestCase() (*gexec.Session, error) {
-	flags := map[string]string{
-		testutil.StartFlagProjectPath: A.AnalysisFolder,
+			horusecConfigFilePathDownloadFromDockerHub := filepath.Join(allTestsCases[idx].Command.Flags[testutil.StartFlagProjectPath], horusecConfigName)
+			RunTestCase(allTestsCases, horusecConfigFilePathDownloadFromDockerHub, "(download from dockerhub)", idx)
+
+			time.Sleep(2 * time.Second)
+
+			if CurrentGinkgoTestDescription().Failed && ginkgoconfig.GinkgoConfig.FailFast {
```

Comment:
we could try asynchronous [ginkgo tests](https://onsi.github.io/ginkgo/#asynchronous-tests), i think we could avoid some `time.Sleep(2 * time.Second)`, i don't think this is a good practice.

Reply:
horusec run your analysis and create .horusec, but in next analysis ginkgo not wait horusec finish completly, then start next analysis insert files into .horusec in new analysisID but this folder was deleted in previus analysis generating error on copy files in pah not exists.